src/jit/lclvars.cpp

```diff
@@ -4150,11 +4149,10 @@ void Compiler::lvaAssignVirtualFrameOffsetsToArgs()
 
     /* Update the argOffs to reflect arguments that are passed in registers */
 
-    noway_assert(codeGen->intRegState.rsCalleeRegArgNum <= MAX_REG_ARG);
```

Comment:
This assert holds for other targets except for Arm64.  How about modifying this assert for ARM64 apropriately?